shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Shrinkers.hs

```diff
@@ -40,33 +41,40 @@ shrinkTx (Tx _b _ws _wm _md) =
 [ Tx b ws wm' | wm' <- shrinkMap shrinkScriptHash shrinkMultiSig wm ]
 -}
 
+-- TODO proper forge shrink!
 shrinkTxBody :: Crypto crypto => TxBody crypto -> [TxBody crypto]
-shrinkTxBody (TxBody is os cs ws tf tl tu md) =
+shrinkTxBody (TxBody is os cs fg ws tf tl tu md) =
   -- shrinking inputs is probably not very beneficial
   -- [ TxBody is' os cs ws tf tl tu | is' <- shrinkSet shrinkTxIn is ] ++
 
   -- Shrink outputs, add the differing balance of the original and new outputs
   -- to the fees in order to preserve the invariant
-  [ TxBody is os' cs ws (tf + (outBalance - outputBalance os')) tl tu md
+  [ TxBody is os' cs fg ws (tf + getAdaAmount (subv outBalance (outputBalance os'))) tl tu md
```

Comment:
That isn't going to work, is it? You're only adding the Ada to the fee, so it won't necessarily balance after this.

Reply:
this needs work